internal/envoy/listener_test.go

```diff
@@ -159,45 +160,104 @@ func TestSocketAddress(t *testing.T) {
 
 func TestDownstreamTLSContext(t *testing.T) {
 	const secretName = "default/tls-cert"
+	const subjectName = "client-subject-name"
+	ca := []byte("client-ca-cert")
+	ciphers := []string{
+		"[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]",
+		"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]",
+		"ECDHE-ECDSA-AES128-SHA",
+		"ECDHE-RSA-AES128-SHA",
+		"ECDHE-ECDSA-AES256-GCM-SHA384",
+		"ECDHE-RSA-AES256-GCM-SHA384",
+		"ECDHE-ECDSA-AES256-SHA",
+		"ECDHE-RSA-AES256-SHA",
+	}
 
-	got := DownstreamTLSContext(secretName, envoy_api_v2_auth.TlsParameters_TLSv1_1, "h2", "http/1.1")
-	want := &envoy_api_v2_auth.DownstreamTlsContext{
-		CommonTlsContext: &envoy_api_v2_auth.CommonTlsContext{
-			TlsParams: &envoy_api_v2_auth.TlsParameters{
-				TlsMinimumProtocolVersion: envoy_api_v2_auth.TlsParameters_TLSv1_1,
-				TlsMaximumProtocolVersion: envoy_api_v2_auth.TlsParameters_TLSv1_3,
-				CipherSuites: []string{
-					"[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]",
-					"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]",
-					"ECDHE-ECDSA-AES128-SHA",
-					"ECDHE-RSA-AES128-SHA",
-					"ECDHE-ECDSA-AES256-GCM-SHA384",
-					"ECDHE-RSA-AES256-GCM-SHA384",
-					"ECDHE-ECDSA-AES256-SHA",
-					"ECDHE-RSA-AES256-SHA",
+	tests := map[string]struct {
+		got  *envoy_api_v2_auth.DownstreamTlsContext
+		want *envoy_api_v2_auth.DownstreamTlsContext
+	}{
+		"TLS context without client authentication": {
+			DownstreamTLSContext(secretName, envoy_api_v2_auth.TlsParameters_TLSv1_1, nil, "", "h2", "http/1.1"),
+			&envoy_api_v2_auth.DownstreamTlsContext{
+				CommonTlsContext: &envoy_api_v2_auth.CommonTlsContext{
```

Comment:
You can hoist this into a variable to reduce copy/paste.

Reply:
Fixed.